api/src/mutations/mint.rs

```diff
@@ -824,6 +845,298 @@ impl Mutation {
+
+        let owner_address = fetch_owner(conn, project_id, blockchain).await?;
+
+        let MetadataJson {
+            metadata_json, uri, ..
+        } = MetadataJson::fetch(mint.id, db).await?;
+
+        let event_key = NftEventKey {
+            id: mint.id.to_string(),
+            user_id: user_id.to_string(),
+            project_id: project_id.to_string(),
+        };
+
+        let creators = mint_creators::Entity::find_by_collection_mint_id(mint.id)
+            .all(conn)
+            .await?;
+
+        let action = if input.compressed {
+            Actions::MintCompressed
+        } else {
+            Actions::Mint
+        };
+
+        let TransactionId(_) = credits
+            .submit_pending_deduction(
+                org_id,
+                user_id,
+                action,
+                collection.blockchain.into(),
+                balance,
+            )
+            .await?;
+
+        match collection.blockchain {
+            BlockchainEnum::Solana => {
+                solana
+                    .event()
+                    .mint_drop(
+                        event_key,
+                        MintDropTransaction::Open(proto::MintMetaplexMetadataTransaction {
+                            metadata: Some(MetaplexMetadata {
+                                owner_address,
+                                name: metadata_json.name,
+                                symbol: metadata_json.symbol,
+                                metadata_uri: uri.unwrap(),
+                                seller_fee_basis_points: mint.seller_fee_basis_points.into(),
+                                creators: creators.into_iter().map(Into::into).collect(),
+                            }),
+                            recipient_address: input.recipient.to_string(),
+                            compressed: input.compressed,
+                            collection_id: collection.id.to_string(),
+                        }),
+                    )
+                    .await?;
+            },
+            BlockchainEnum::Ethereum | BlockchainEnum::Polygon => {
+                return Err(Error::new("blockchain not supported as this time"));
+            },
+        };
+
+        let mut mint_am: collection_mints::ActiveModel = mint.into();
+        mint_am.creation_status = Set(CreationStatus::Pending);
+        let mint = mint_am.update(conn).await?;
+
+        let mint_history_am = mint_histories::ActiveModel {
+            mint_id: Set(mint.id),
+            wallet: Set(input.recipient),
+            collection_id: Set(collection.id),
+            tx_signature: Set(None),
+            status: Set(CreationStatus::Pending),
+            created_at: Set(Utc::now().into()),
+            ..Default::default()
+        };
+
+        mint_history_am.insert(conn).await?;
+
+        Ok(MintQueuedPayload {
+            collection_mint: mint.into(),
+        })
+    }
+
+    async fn mint_random_queued(
```

Comment:
```suggestion
    async fn mint_random_queued_to_drop(
```

Just to keep the naming more consistent.